pandas/core/array_algos/quantile.py

```diff
@@ -0,0 +1,76 @@
+from typing import Sequence, Union
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.common import is_list_like
+
+from pandas.core.nanops import nanpercentile
+
+
+def quantile_with_mask(
+    values: np.ndarray,
+    mask: np.ndarray,
+    fill_value,
+    qs: Union[float, Sequence[float]],
+    interpolation: str,
+    axis: int,
+) -> np.ndarray:
+    """
+    Compute the quantiles of the given values for each quantile in `qs`.
+
+
+    Parameters
+    ----------
+    values : np.ndarray
+        For ExtensionArray, this is _values_for_factorize()[0]
+    mask : np.ndarray[bool]
+        mask = isna(values)
+        For ExtensionArray, this is computed before calling _value_for_factorize
+    fill_value : Scalar
```

Comment:
Could also call this `na_value` ? (like nanpercentile does) 

(I know we use both fill_value and na_value in many places, and somewhat interchangeably, but here I personally find na_value clearer)

Reply:
I thought about this and decided on fill_value on the theory that "na_value" means "the value that, when we see it, indicates we have an NA" and "fill_value" means "the value that we use when we need to fill in an NA value".

e.g. we get here with fill_value=iNaT, which would be weird to have as an na_value